assigner.py

```diff
@@ -20,173 +22,257 @@
                     raise
 
 
 def assign(args):
-    if args.dry_run:
-        raise NotImplementedError("'--dry-run' is not implemented")
-    if args.student:
+    """Creates homework repositories for an assignment for each student
+    in the roster.
+    """
+    hw_name = args.name
+    branch = args.branch
+    section = args.section
+    dry_run = args.dry_run
+    force = args.force
+    verbose = args.verbose
+    target = args.student  # used if assigning to a single student
+
+    if target:
         raise NotImplementedError("'--student' is not implemented")
 
     with config(args.config) as conf, tempfile.TemporaryDirectory() as tmpdirname:
-        base = BaseRepo(conf['gitlab-host'], conf['namespace'], args.name, conf['token'])
-        if args.branch:
-            base.clone_to(tmpdirname, args.branch)
+        host = conf['gitlab-host']
+        namespace = conf['namespace']
+        token = conf['token']
+        semester = conf['semester']
+        if section:
+            roster = [s for s in conf['roster'] if s['section'] != section]
         else:
-            base.clone_to(tmpdirname)
-
-        count = 0
-        for student in conf['roster']:
-            if args.section and student['section'] != args.section:
-                continue
-
+            roster = conf['roster']
+
+        base = BaseRepo(host, namespace, hw_name, token)
+        if not dry_run:
+            base.clone_to(tmpdirname, branch)
+
+        actual_count = 0  # Represents the number of repos actually pushed to
+        s_count = len(roster)
+        print("Assigning {} to {} student{} in {}...".format(
+            hw_name, s_count,
+            "s" if s_count != 1 else "",
+            "section " + section if section else "any section")
+        )
+        for i, student in enumerate(roster):
+            username = student['username']
+            s_section = student['section']
+            full_name = StudentRepo.name(semester, s_section, hw_name, username)
+            repo = StudentRepo(host, namespace, full_name, token)
+
+            # Is it always necessary to log this?  Feels wrong.
+            logging.warning(
+                "Student repository {} already exists.".format(repo.name)
+            ) if verbose else None
             try:
-                name = StudentRepo.name(conf['semester'], student['section'], args.name, student['username'])
-
-                repo = StudentRepo(conf['gitlab-host'], conf['namespace'], name, conf['token'])
-                repo.info
-
-                logging.warning("Student repository %s already exists.", repo.name)
-
-                if args.force:
-                    logging.warning("Deleting...")
-                    repo.delete()
-                    repo = StudentRepo.new(base, conf['semester'], student['section'], student['username'], conf['token'])
-
-                    if args.branch:
+                if force:
+                    logging.warning("Deleting...") if verbose else None
```

Comment:
I think something as important as deleting a student repo should result in output...assigner isn't trying to be `rm`, so we can make a little noise!